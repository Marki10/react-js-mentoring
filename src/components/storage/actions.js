export const setSortBy = (value) => {
  return {
    type: value
  };
};

export const setSearchBy = (value) => {
  return {
    type: value
  };
};

export const searchQuery = (value) => {
  return {
    type: 'SET_SEARCH_QUERY',
    payload: value
  };
};

export const getMovies = (query) => {
  return {
    type: 'GET_MOVIES',
    query: query
  };
};
