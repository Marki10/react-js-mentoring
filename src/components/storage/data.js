export const movies = [
    {
        "unit":883,
        "show_id":70299043,
        "show_title":"Attack on Titan",
        "release_year":"2013",
        "rating":"4.6",
        "category":"Anime",
        "show_cast":"Yuki Kaji, Yui Ishikawa, Marina Inoue, Daisuke Ono, Hiro Shimono, Hiroshi Kamiya, Keiji Fujiwara, Kish\u00f4 Taniyama, Romi Park, Ryota Ohsaka",
        "director":"",
        "summary":"For over a century, people have been living behind barricades to block out the giant Titans that threaten to destroy the human race. When a Titan destroys his hometown, young Eren Yeager becomes determined to fight back.",
        "poster":"http://cdn-2.nflximg.com/en_us/boxshots/ghd/70153391.jpg",
        "mediatype":1,
        "runtime":"24 min"
    },
    {
        "unit":6596,
        "show_id":70153391,
        "show_title":"The Boondocks",
        "release_year":"2005",
        "rating":"4",
        "category":"Comedies\n",
        "show_cast":"Regina King, John Witherspoon, Cedric Yarbrough, Gary Anthony Williams, Jill Talley, Gabby Soleil",
        "director":"",
        "summary":"Based on the comic strip by Aaron McGruder, this satirical animated series follows the socially conscious misadventures of Huey Freeman, a preternaturally smart 10-year-old who relocates from inner-city Chicago to the suburbs.",
        "poster":"http://cdn-2.nflximg.com/en_us/boxshots/ghd/70153391.jpg",
        "mediatype":1
    },
    {
        "unit":6596,
        "show_id":70153391,
        "show_title":"The Boondocks",
        "release_year":"2005",
        "rating":"4",
        "category":"Comedies\n",
        "show_cast":"Regina King, John Witherspoon, Cedric Yarbrough, Gary Anthony Williams, Jill Talley, Gabby Soleil",
        "director":"",
        "summary":"Based on the comic strip by Aaron McGruder, this satirical animated series follows the socially conscious misadventures of Huey Freeman, a preternaturally smart 10-year-old who relocates from inner-city Chicago to the suburbs.",
        "poster":"http://cdn-2.nflximg.com/en_us/boxshots/ghd/70153391.jpg",
        "mediatype":1
    },
    {
        "unit":883,
        "show_id":70299043,
        "show_title":"Attack on Titan",
        "release_year":"2013",
        "rating":"4.6",
        "category":"Anime",
        "show_cast":"Yuki Kaji, Yui Ishikawa, Marina Inoue, Daisuke Ono, Hiro Shimono, Hiroshi Kamiya, Keiji Fujiwara, Kish\u00f4 Taniyama, Romi Park, Ryota Ohsaka",
        "director":"",
        "summary":"For over a century, people have been living behind barricades to block out the giant Titans that threaten to destroy the human race. When a Titan destroys his hometown, young Eren Yeager becomes determined to fight back.",
        "poster":"http://cdn-2.nflximg.com/en_us/boxshots/ghd/70153391.jpg",
        "mediatype":1,
        "runtime":"24 min"
    },
    {
        "unit":6596,
        "show_id":70153391,
        "show_title":"The Boondocks",
        "release_year":"2005",
        "rating":"4",
        "category":"Comedies\n",
        "show_cast":"Regina King, John Witherspoon, Cedric Yarbrough, Gary Anthony Williams, Jill Talley, Gabby Soleil",
        "director":"",
        "summary":"Based on the comic strip by Aaron McGruder, this satirical animated series follows the socially conscious misadventures of Huey Freeman, a preternaturally smart 10-year-old who relocates from inner-city Chicago to the suburbs.",
        "poster":"http://cdn-2.nflximg.com/en_us/boxshots/ghd/70153391.jpg",
        "mediatype":1
    },
    {
        "unit":6596,
        "show_id":70153391,
        "show_title":"The Boondocks",
        "release_year":"2005",
        "rating":"4",
        "category":"Comedies\n",
        "show_cast":"Regina King, John Witherspoon, Cedric Yarbrough, Gary Anthony Williams, Jill Talley, Gabby Soleil",
        "director":"",
        "summary":"Based on the comic strip by Aaron McGruder, this satirical animated series follows the socially conscious misadventures of Huey Freeman, a preternaturally smart 10-year-old who relocates from inner-city Chicago to the suburbs.",
        "poster":"http://cdn-2.nflximg.com/en_us/boxshots/ghd/70153391.jpg",
        "mediatype":1
    }
];
