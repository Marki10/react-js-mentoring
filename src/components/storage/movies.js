import axios from 'axios';

const movieReducer = (state = [], action) => {
    let movies = [];
    switch (action.type) {
        case "GET_MOVIES":
            movies = action.payload;
            return movies;
    default:
        return state
    }

    return state;
}

export default movieReducer;
