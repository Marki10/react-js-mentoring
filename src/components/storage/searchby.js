import {createStore} from "redux";

const searchByReducer = (state = "title", action) => {
    switch (action.type) {
        case 'title':
            return state = 'title';

        case 'director':
            return state = 'director';

        default:
            return state;
    }
}

export default searchByReducer;
