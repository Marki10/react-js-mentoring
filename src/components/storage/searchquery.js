const searchQueryReducer = (state = [], action) => {
    let result;
    switch (action.type) {
        case 'SET_SEARCH_QUERY':
            return state = action.payload;

        default:
            return state;
    }
}

export default searchQueryReducer;