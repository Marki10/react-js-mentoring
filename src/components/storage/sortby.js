import {createStore} from "redux";

const sortByReducer = (state = "release_date", action) => {
    switch (action.type) {
        case 'release_date':
            return state = "release_date";

        case 'vote_average':
            return state = "vote_average";

        default:
            return state;
    }
}

export default sortByReducer;
