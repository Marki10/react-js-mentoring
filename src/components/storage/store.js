import { combineReducers } from 'redux';
import movieReducer from './movies';
import sortByReducer from './sortby';
import searchByReducer from './searchby';
import searchQueryReducer from './searchquery';

const storeApp = combineReducers({
    movieReducer,
    searchByReducer,
    sortByReducer,
    searchQueryReducer
})

export default storeApp;
