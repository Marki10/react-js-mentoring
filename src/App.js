// Core
import React, { Component } from 'react';
import { connect } from "react-redux";

// Components
import Header from './components/header';
import FilmsList from './components/filmslist';
import Footer from './components/footer';

// Actions
import { setSortBy, getMovies } from './components/storage/actions';

// Sass / Css
import './App.sass';

class App extends Component {
    constructor (props) {
        super(props);
    }
    
    render() {
        return (
            <div className="App">
                <Header/>
                <FilmsList films={this.props.movies.results} sortby={this.props.sortby}/>
                <Footer />
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        movies: state.movieReducer,
        sortby: state.sortByReducer,
        searchby: state.searchByReducer,
        searchquery: state.searchQueryReducer
    };
};

const mapDispatchToProps = (dispatch) => ({
    setSortBy: (value) => {
        dispatch(setSortBy(value));
    },
    getMovies: (query) => {
        dispatch(getMovies(query));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
