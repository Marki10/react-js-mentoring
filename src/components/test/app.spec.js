import React, { Component } from 'react';
import { connect } from "react-redux";
import App from '../../App';
import Header from '../header';
import FilmsList from '../filmslist';
import Footer from '../footer';
import { setSortBy, getMovies } from '../storage/actions';
const ReactTestRenderer = require('react-test-renderer');

describe('App', () => {
    it('Render app', () => {
        const component = ReactTestRenderer.create(
            connect(<App />)
        );
        const testTemp = component.toJSON();
        expect(testTemp).toMatchSnapshot();
    })
});