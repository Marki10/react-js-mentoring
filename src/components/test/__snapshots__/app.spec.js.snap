exports[`App Render app 1`] = `null`;
